06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * The Inverted Index class.
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.globalIndex = {};
+  }
+
+  /**
+   * A method that converts string into an array
+   * @param  {string} word the string  from the "text" key in the JSON
+   * @return {Array} containing String.
+   */
+  static tokenize(word) {
+    const terms = word.replace(/[^\w\s]/gi, ' ')
+      .match(/\w+/g);
+    return terms;
+  }
+
+  /**
+   * A method that filters the array passed into it for unique words
+   * @param {Array} words Array of strings
+   * @returns {Array} Returns array of unique words
+   */
+  static uniqueWords(words) {
+    const unique = [...new Set(words)];
+    return unique;
+  }
+
+  /**
+   * A method that tokenizes an object and gets the unique terms in the object
+   * @param  {Object} file the uploaded JSON file object
+   * @return {Array}  of the unique terms in the file
+   */
+  static getText(file) {
+    let newText = '';
+    const space = ' ';
+    Object.keys(file)
+      .map((index) => {
+        newText += `${file[index].text} ${space}`;
+        return newText;
+      });
+    const terms = InvertedIndex.tokenize(newText);
+    const uniqueTerms = InvertedIndex
+      .uniqueWords(terms)
+      .map(x => x.toLowerCase());
```

Comment:
Thank you for pointing this out, I have rectified this issue, by using the correct variable naming convention.